app/src/main/java/com/stevesoltys/seedvault/transport/backup/BackupCoordinator.kt

```diff
@@ -211,13 +231,30 @@ internal class BackupCoordinator(
         data: ParcelFileDescriptor,
         flags: Int
     ): Int {
-        cancelReason = UNKNOWN_ERROR
+        state.cancelReason = UNKNOWN_ERROR
         val packageName = packageInfo.packageName
-        // K/V backups (typically starting with package manager metadata)
+        // K/V backups (typically starting with package manager metadata - @pm@)
         // are scheduled with JobInfo.Builder#setOverrideDeadline() and thus do not respect backoff.
-        // We need to reject them manually when now is not a good time for a backup.
-        if (packageName == MAGIC_PACKAGE_MANAGER && getBackupBackoff() != 0L) {
-            return TRANSPORT_PACKAGE_REJECTED
+        // We need to reject them manually when we can not do a backup now.
+        // What else we tried can be found in: https://github.com/stevesoltys/seedvault/issues/102
+        if (packageName == MAGIC_PACKAGE_MANAGER) {
+            if (!settingsManager.canDoBackupNow()) {
+                // Returning anything else here (except non-incremental-required which re-tries)
+                // will make the system consider the backup state compromised
```

Comment:
This also means we should test this commit / set of changes a bit more.

Scenarios:
* Backup to USB, plug in routinely, see if still works as expected, would be same as before.
* Backup to USB, then don't plug in USB for a few days, see if it still works afterward - this should no longer create a new "restore set"
* Backup to internal storage, should be unaffected
* Backup to nextcloud, then no internet - this is handled by Seedvault
* Backup to nextcloud, then server unavailable - I'm curious as to what would happen here.

Reply:
>  this should no longer create a new "restore set"

we already don't create new restore sets anymore when the get told to re-initialize, but we wipe all data as instructed. When testing this, you will need to observe logcat to notice a difference to before.

Reply:
Most of what I listed is covered by https://github.com/stevesoltys/seedvault/issues/152